app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/models/AlarmByRuleListApiModel.java

```
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+package com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.microsoft.azure.iotsolutions.devicetelemetry.services.models.AlarmServiceModel;
+import com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.Version;
+import org.joda.time.DateTime;
+
+import java.util.*;
+
+public class AlarmByRuleListApiModel {
+    private final ArrayList<AlarmByRuleApiModel> items;
+
+    public enum Status {
+        open  (3),
+        acknowledged(2),
+        closed   (1)
+        ;
+
+        private final int statusCode;
+
+        private Status(int statusCode) {
+            this.statusCode = statusCode;
+        }
+    }
+
+    public AlarmByRuleListApiModel(final ArrayList<AlarmServiceModel> alarms) {
+        this.items = new ArrayList<>();
+        if (alarms != null) {
+            HashMap<String, AlarmByRuleApiModel> map = new HashMap<>();
+            for (AlarmServiceModel alarm : alarms) {
+                String id = alarm.getRuleId();
+                if(!map.containsKey(id)) {
+                    map.put(id, new AlarmByRuleApiModel(1, alarm.getStatus(), alarm.getDateModified(), alarm));
+                } else {
+                    AlarmByRuleApiModel alarmByRule = map.get(id);
+                    alarmByRule.setTotalCount(alarmByRule.getTotalCount() + 1);
+                    DateTime occurred = alarm.getDateCreated();
+                    if(DateTime.parse(alarmByRule.getLastOccurred()).isBefore(occurred)) {
+                        alarmByRule.setLastOccurred(occurred);
+                    }
+
+                    String savedStatus = alarmByRule.getStatus();
+                    String newStatus = alarm.getStatus();
+                    if(Status.valueOf(savedStatus).statusCode < Status.valueOf(newStatus).statusCode) {
```

could add a comment here? I'm not totally sure about the intent, are we denying a request of some type (in which case we should throw an error) ?